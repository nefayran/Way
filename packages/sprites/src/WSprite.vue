<template>
  <svg
    class="w-sprite"
    :width="width"
    :height="height"
    aria-hidden="true"
    :style="`color: ${color}`"
  >
    <use :xlink:href="symbolId" />
  </svg>
</template>

<script lang="ts" setup>
import { withDefaults, computed } from "vue";

interface Props {
  prefix?: string;
  name: string;
  color?: string;
  width?: string;
  height?: string;
}

const props = withDefaults(defineProps<Props>(), {
  prefix: "sprite",
  color: "#333",
  width: "128",
  height: "128",
});

const symbolId = computed(() => `#${props.prefix}-${props.name}`);
</script>
