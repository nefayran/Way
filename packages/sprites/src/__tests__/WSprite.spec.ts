import { mount } from "@vue/test-utils";
import WSprite from "../WSprite.vue";

describe("WSprite.vue", () => {
  test("should have an w-sprite class", () => {
    const wrapper = mount(WSprite, {
      propsData: {
        prefix: "test",
        name: "color",
        color: "#fff",
        width: "50",
        height: "60",
      },
    });
    expect(wrapper.classes()).toContain("w-sprite");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
