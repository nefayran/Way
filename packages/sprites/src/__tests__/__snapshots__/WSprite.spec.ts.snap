// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WSprite.vue should have an w-sprite class 1`] = `
<svg class="w-sprite" width="50" height="60" aria-hidden="true" style="color: rgb(255, 255, 255);">
  <use xlink:href="#test-color"></use>
</svg>
`;
